### ISSUE TITLE
Extract multiple files from archive

### ISSUE THREAD

**Issue by Entze** [houseabsolute/ubi#68]
When installing the [Helix-Editor](https://github.com/helix-editor/helix/) with ubi, the binary is downloaded correctly, however the _runtime_ directory, containing tree-sitter definitions, themes and the tutor are missing.

As a user I would like to specify a path (maybe multiple) of resources that are extracted along side the single binary, such that static configuration files are not missing.

Let me know if you need any more insights or help.

**Comment by autarch** [comment]
This is something I've thought about a bit. I think maybe the best option is just to have a way to extract _either_ just one file (the executable) or to just dump all the contents of the archive under one directory.

**Comment by Entze** [comment]
That might be a good approach in case a project provides multiple binaries as well.

It might be beneficial to gather usecases/projects where such a setting would be useful. I will look through the tools I use and gather usecases if you don't mind.

**Comment by autarch** [comment]
> It might be beneficial to gather usecases/projects where such a setting would be useful. I will look through the tools I use and gather usecases if you don't mind.

Yes, please. I'd be interested in knowing more about these use cases.

**Comment by Entze** [comment]
A review of the apps/tools I use, showed ubi in relation to the following situations:

- works out of the box (most Rust or Go binaries)
- optional files missing (same directory) ([helix](https://github.com/helix-editor/helix))
- optional files missing (specific directories, in this case shell completions) ([gum](https://github.com/charmbracelet/gum), [glow](https://github.com/charmbracelet/glow))
- only distribution/operating system specific package format available ([flox](https://github.com/flox/flox/))
- Not compiled yet ([dateutils](https://github.com/hroptatyr/dateutils/))

Importantly, to reiterate, I'm not asking for the scope to be expanded to cover all those cases.

**Comment by kabouzeid** [comment]
I'd also love to have that functionality. Right now, I rely on hardcoded `wget`, `tar`, and `ln` commands in my dotfiles to download GitHub releases, extract their contents, and link binaries, completions, and man pages to the appropriate locations. It would be great if `ubi` could handle finding the correct links and extraction steps. I’d still manually link completions and man pages to the right places since that process often varies across packages and may be challenging to automate fully in `ubi`.

For reference: [install-binaries.conf.yaml](https://github.com/kabouzeid/dotfiles/blob/main/install-binaries.conf.yaml)

**Comment by autarch** [comment]
@kabouzeid Just to clarify. I imageint that running `ubi --project chmln/sd --extract-in some/dir` might give you a tree like this:

```
sd
├── completions
│   └── sd.fish
├── sd
└── sd.1
```

Is this what you want?

Note that `--extract-in` is just something I made up right now. I'd want to think more about the exact details of how to specify this "extract a tree" mode would work.

**Comment by kabouzeid** [comment]
yes, exactly. another way to design the API would be to keep the current `--in` flag, and have additional `--binary-only` / `--no-binary-only` flags.

**Comment by autarch** [comment]
> yes, exactly. another way to design the API would be to keep the current `--in` flag, and have additional `--binary-only` / `--no-binary-only` flags.

Yeah, I was thinking of something like this, but reversed, like `--extract-all`. Then we don't need a `	--no-extract-all`, because that's the default behavior.

**Comment by kabouzeid** [comment]
fully agree

**Comment by autarch** [comment]
This is in the v0.4.0 release, which I just did.
